src/Functions/HasTokenImpl.h

```diff
@@ -46,10 +47,24 @@ struct HasTokenImpl
         /// The current index in the array of strings.
         size_t i = 0;
 
-        TokenSearcher searcher(pattern.data(), pattern.size(), end - pos);
+        std::optional<TokenSearcher> searcher;
+        try
+        {
+            searcher.emplace(pattern.data(), pattern.size(), end - pos);
+            if (null_map)
+                null_map->getData().resize_fill(haystack_offsets.size(), false);
+        }
+        catch (...)
+        {
+            if (!null_map)
+                throw;
+
+            null_map->getData().resize_fill(haystack_offsets.size(), true);
```

Comment:
As I see by the stacktrace I posted above, the problem is here
https://github.com/ClickHouse/ClickHouse/blob/fdea042991f2a020f6ed18d860a67dea8438403d/src/Columns/FilterDescription.cpp#L80-L83
and the reason is probably that the size you pass to `resize_fill` is incorrect. It should same as here `vec_res.resize(column_haystack->size());` but as I understand `haystack_offsets.size()` is not the same. Can you please check this place?

Reply:
Great catch, thanks! I was just pondering how a seemingly irrelevant piece of code might break, but what you're suggesting is very much on the right track. Let me check.

Reply:
Eventually, the issue came down to the fact that `res` was uninitialized.